system-x/services/ldap/src/main/java/software/tnb/ldap/validation/LdapValidation.java

```diff
@@ -0,0 +1,44 @@
+package software.tnb.ldap.validation;
+
+import software.tnb.common.validation.Validation;
+
+import com.unboundid.ldap.sdk.LDAPConnection;
+import com.unboundid.ldap.sdk.LDAPException;
+import com.unboundid.ldap.sdk.SearchRequest;
+import com.unboundid.ldap.sdk.SearchResult;
+import com.unboundid.ldap.sdk.SearchResultEntry;
+import com.unboundid.ldap.sdk.SearchScope;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LdapValidation implements Validation {
+
+    private String ldapUsername;
+    private String ldapPassword;
+    private String ldapUrl;
+
+    public LdapValidation(String ldapUsername, String ldapPassword, String ldapUrl) {
+        this.ldapUsername = ldapUsername;
+        this.ldapPassword = ldapPassword;
+        this.ldapUrl = ldapUrl;
+    }
+
+    public List<String> getEntries(String objectClass) throws LDAPException {
```

Comment:
rather than throwing a checked exception, wrap the related calls in try-catch block and throw an unchecked runtimeexception

Reply:
Done